### PR TITLE
Gstreamer plugin: Enhance profile setting and fix some issues

### PR DESCRIPTION
**Pull request by Yeongjin-Jeong**
* Fix deadlock on encoder initializing failure
Don't miss unlock before returning.

* Ensure that slice type is initialized to default value
Don't keep the previous slice type. This may request an unintended slice type.

* Enhance profile setting with requested profile by peer
SVT-HEVC encoder does not yet support auto profile selection, So try
all profiles allowed by peer until SVT-HEVC encoder can accept one of them.

* Consider subset relations of profiles when setting srcpad caps
Relaxing the profile condition since libSvtHevcEnc can generate
wrong bitstream indication for conformance to profile than requested one.

* Don't use deprecated latency_mode parameter
The SVT-HEVC encoder does not support latency_mode parameter since 1.4.0.

* Remove unused headers

* Bump libsvthevcenc requirement to 1.4.1

* Disable codeVpsSpsPps flag depending on version

* Fix merge conflict in latest Gstreamer

### CODE REVIEW

**Comment by Yeongjin-Jeong** [comment]
Hi @tianjunwork,
In some test cases, I found negotiation failure bugs with this MR. So I will patch it and create MR again.

File: gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch [hunk 1]
```diff
@@ -1914,7 +2021,6 @@ index 0000000..283027f
 +  guint qp_min;
 +  gboolean scene_change_detection;
 +  guint tune;
-+  guint latency_mode;
 +  guint base_layer_switch_mode;
```
[the remark this answers is not in the document]
Comment (Yeongjin-Jeong) [reply gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch]: Inside Lib doesn't seems to allow latencyMode to be non-zero. (https://github.com/OpenVisualCloud/SVT-HEVC/blob/master/Source/Lib/Codec/EbEncHandle.c#L3043) 
And if I set latencyMode to 1, Inside Lib doesn't force it to 0 and just copy to context (https://github.com/OpenVisualCloud/SVT-HEVC/blob/master/Source/Lib/Codec/EbEncHandle.c#L2438), so I failed to initialize the encoder.

**Comment by Yeongjin-Jeong** [comment]
Hi @tianjunwork, I pushed new commits for fixing negotiation issues. And fixed some code that could cause minor bugs.
BTW, Do you have additional release plan in with patch level like v1.4.1? If not, now I want to work on upstream this git-path to gstreamer. what do you think about?

**Comment by tianjunwork** [comment]
We don't have plan to tag minor release. That would be great for you working on upstream. Thank you very much for your contribution. Could you fire an issue that we track all the discussion during upstreaming?

**Comment by Yeongjin-Jeong** [comment]
> We don't have plan to tag minor release.
Ok, then https://github.com/OpenVisualCloud/SVT-HEVC/pull/250 might be reverted on this git-patch during working on upstream. This API is not compatible with the previous version (e.g. v1.4.0).
> Could you fire an issue that we track all the discussion during upstreaming?
Does that mean create issue at this repo? or at gst repo?

Thanks :)

**Comment by tianjunwork** [comment]
@Yeongjin-Jeong if adding a minor release tag helps, we can do that:)
Yes, creating issue an issue at this repo: https://github.com/OpenVisualCloud/SVT-HEVC/issues

**Comment by tianjunwork** [comment]
@Yeongjin-Jeong I will be out of office for almost 30 days. Please go ahead up streaming to gst. If you need any help from SVT side, I will try to coordinate. Thank you:)

**Comment by Yeongjin-Jeong** [comment]
> if adding a minor release tag helps, we can do that:)

Since gstreamer is strict for API compatibility, it would be nice if the library version was guaranteed for API usage changes.

> I will be out of office for almost 30 days. Please go ahead up streaming to gst. If you need any help from SVT side, I will try to coordinate.

Sure, I'll start working upstream. :) BTW, I have a question. In the upstream process, gstreamer might ask me to change the naming called svthevc to something like svth265 or svt265. What do you think? Is it absolutely impossible?

**Comment by tianjunwork** [comment]
@Yeongjin-Jeong, for the naming issue, thanks for the heads up. If gst community insists the name of  `svth265` , we can go with it.

**Comment by Yeongjin-Jeong** [comment]
Hi @tianjunwork, Recent working branch merged into the gstreamer master branch. (https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/merge_requests/731)
That branch has more revisions, so I'm going to discard this MR and create a new one.
And the new MR will include some commits of the independent gstreamer plugin for property unification. (There are several properties that have been changed to the gstreamer style.)

**Comment by Yeongjin-Jeong** [comment]
@tianjunwork, I have one question. Now that this plugin has been merged into the gstreamer master branch, Would it be right to keep the git-patch in this repo? If gstreamer users see this patch, it can be confusing.
How about removing the git patch from this repo and just keeping the interface synchronized with the independent plugin?

**Comment by tianjunwork** [comment]
Thank you so much for upstreaming gst SVT-HEVC plugin! That's really a good news. I do agree the gst patch in this repo can be removed. Could you work on the PR? We will keep an eye on the interface to keep the independent one updated. Thanks again for your effort. Wish you a happy new year 😊

**Pull request closed without merging**